beam/www/beam/index.ts

```diff
@@ -94,6 +107,19 @@ const router = createRouter({
 	routes,
 })
 
+router.beforeEach((to, from, next) => {
```

Comment:
I think we want to detect if the Vite server is running and opt out of this for development. This may not be a great idea for security purposes, so lets try to pen test it with some console hacking. May need @Alchez work on #121

Reply:
@agritheory @MyuddinKhatri I think this _may_ be fine this way since the entrypoint for the development setup (`/dev.html`) is separate from the production setup (`/beam`) anyway.

But if there's a part of the development setup that would need some production setup, then we can look at the Vite metadata. Vite [recommends](https://vitejs.dev/guide/env-and-mode) checking against the `import.meta.env.DEV` and `import.meta.env.PROD` keys for development vs production setups.